src/generation/chat/mod.rs

```diff
@@ -99,15 +97,69 @@ impl Ollama {
 
 #[cfg(feature = "chat-history")]
 impl Ollama {
+    #[cfg(feature = "stream")]
+    pub async fn send_chat_messages_with_history_stream<S: Into<String> + Clone>(
+        &mut self,
+        mut request: ChatMessageRequest,
+        history_id: S,
```

Comment:
perhaps it's better to use `history_id: impl ToString` to simplify generic type parameter here. And since `String` type has already implemented `Clone` trait, `Clone` in generic type parameter can be removed.

Reply:
Would you mind testing it out on your end, please? If we want to change parameter's type only in this place, I'm getting error : 
```
the trait `Clone` is not implemented for `impl ToString`
```

Reply:
like this:
```rust
pub async fn send_chat_messages_with_history_stream(
    &mut self,
    mut request: ChatMessageRequest,
    history_id: impl ToString,
) -> crate::error::Result<ChatMessageResponseStream> {
    use async_stream::stream;
    use tokio_stream::StreamExt;
  
    // Use .to_string().clone() instead.
    let id_copy = history_id.to_string().clone();
    ...
}
```

Reply:
@ZBcheng  Thanks! I don't know why my IDE didn't show anything other than `history_id.type_id`. Now using your method everywhere :)